src/components/Reports/Reports.js

```diff
@@ -142,37 +139,37 @@ const Reports = ({ dependencies: { datahubClient } }) => {
                     today={state.dailyView}
                     emailFilter={'with_email'}
                     visits={totalVisits.withEmail}
-                    loading={totalVisits.loading}
+                    loading={state.loading || totalVisits.loading}
                   />
                 </div>
-                {!state.dailyView ? (
+                {!state.dailyView && (
                   <div className="col-sm-12 m-b-24">
                     <ReportsDailyVisits
-                      domainName={state.domainSelected.name}
+                      domainName={state.domainSelected?.name}
                       dateFrom={state.dateFrom}
                       dateTo={state.dateTo}
                     />
                   </div>
-                ) : null}
+                )}
```

Comment:
Are you sure about replacing the ternary operator with a boolean operation?

What happens when it evaluates as false?

See https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx

Reply:
Also, these changes are clearly a refactor, it is not related to changing behavior. 

In my opinion, it is better to split these kinds of changes of behavioral changes in different commits or even different PRs. See https://medium.com/@kentbeck_7670/bs-changes-e574bc396aaa

Maybe, the [refactoring is required to do the behavior change](https://twitter.com/KentBeck/status/250733358307500032). In this case, you can have a single PR with two commits: the first for the refactoring and the second for the behavior.

Reply:
[More details](https://reactjs.org/docs/jsx-in-depth.html#booleans-null-and-undefined-are-ignored)

![image](https://user-images.githubusercontent.com/84402180/133430390-fe86fd6a-2e44-4725-8f0e-f75708de3bdb.png)

Reply:
So, Kent Dodds worries are wrong?

Reply:
In my opinion, this discussion is not related to the change with the spinners. 

This change should have its own PR. Because in another way, tomorrow I apply another change and mixed with it I refactor it again based on Doods opinion 😛, the next day you change it again and so.